packages/tcc/build.sh

```diff
@@ -30,81 +33,26 @@ termux_step_post_get_source() {
 	fi
 }
 
-termux_step_pre_configure() {
-	cd $TERMUX_PKG_BUILDDIR
-	rm -rf _bin
-	mkdir _bin
-	cd _bin
-	local _ar="$TERMUX_HOST_PLATFORM-ar"
-	cat > "${_ar}" <<-EOF
-		#!$(command -v sh)
-		exec $(command -v $AR) "\$@"
-	EOF
-	chmod 0700 "${_ar}"
-	export PATH="$(pwd):$PATH"
-}
-
 termux_step_configure() {
-	unset CFLAGS CXXFLAGS
-
-	if [ "${TERMUX_ARCH}" = "arm" ] || [ "${TERMUX_ARCH}" = "i686" ]; then
-		ELF_INTERPRETER_PATH="/system/bin/linker"
-		ANDROID_LIB_PATH="/system/lib:/system/vendor/lib"
-	else
-		ELF_INTERPRETER_PATH="/system/bin/linker64"
-		ANDROID_LIB_PATH="/system/lib64:/system/vendor/lib64"
-	fi
+	# override default step
+        echo "running on : $(uname -a)"
 }
 
 termux_step_make() {
 	(
-		sysinc=
-		otherinc=
-		for d in $(echo | $CC -E -x c - -v 2>&1 | \
-				sed -n '/^#include <...> search/,/^End/p' | \
-				grep '^\s'); do
-			p="$(readlink -f "${d}"):"
-			if [[ "${d}" = */sysroot/usr/* ]]; then
-				sysinc+="${p}"
-			else
-				otherinc+="${p}"
-			fi
-		done
-		sysinc="${sysinc}${otherinc%:}"
 		unset CC CFLAGS LDFLAGS
-		./configure \
-			--prefix="/tmp/tcc.host" \
-			--cpu="${TERMUX_ARCH}" \
-			--sysincludepaths="${sysinc}"
-		make -j $TERMUX_MAKE_PROCESSES tcc
-		mv -f tcc tcc.host
-		make distclean
+		# make a cross-tcc and use it to compile the runtime
+		./configure --targetos="Termux" --cpu="$TERMUX_ARCH" \
+			--sysroot="${TERMUX_STANDALONE_TOOLCHAIN}/sysroot/usr"
+		make -j $TERMUX_MAKE_PROCESSES tcc libtcc1.a
 	)
 
-	./configure \
-		--prefix="$TERMUX_PREFIX" \
-		--cross-prefix="${CC//clang}" \
-		--cc="clang" \
-		--cpu="$TERMUX_ARCH" \
-		--disable-rpath \
-		--elfinterp="$ELF_INTERPRETER_PATH" \
-		--crtprefix="$TERMUX_PREFIX/lib/tcc/crt" \
-		--sysincludepaths="$TERMUX_PREFIX/include/$TERMUX_HOST_PLATFORM:$TERMUX_PREFIX/include:$TERMUX_PREFIX/lib/tcc/include" \
-		--libpaths="$TERMUX_PREFIX/lib:$TERMUX_PREFIX/lib/tcc:$ANDROID_LIB_PATH"
-
-	mv tcc.host tcc
-	touch -d "next minute" tcc
-	make -j ${TERMUX_MAKE_PROCESSES} libtcc1.a
-
-	rm -f tcc
+	# now cross-compile tcc for target
+	./configure --targetos=Termux --cpu="$TERMUX_ARCH"
 	make -j ${TERMUX_MAKE_PROCESSES} tcc
+	make doc
 }
 
-termux_step_post_make_install() {
-	mkdir -p "${TERMUX_PKG_MASSAGEDIR}/${TERMUX_PREFIX}"/lib/tcc/crt
-	for file in crtbegin_dynamic.o crtbegin_so.o crtend_android.o crtend_so.o; do
-		install -Dm600 \
-			"${TERMUX_STANDALONE_TOOLCHAIN}/sysroot/usr/lib/$TERMUX_HOST_PLATFORM/$TERMUX_PKG_API_LEVEL/$file" \
-			"${TERMUX_PREFIX}/lib/tcc/crt/$file"
-	done
+termux_step_make_install() {
+	make install
```

Comment:
Please drop this function as well, the default `termux_step_make_install` should work fine, no need to override it

Reply:
Thanks, done.